### PR TITLE
Add rebasing to DDS fuzz tests

### PR DESCRIPTION
**Pull request by andre4i**
## Description

This should help catch eventual consistency bugs in DDSes from enabling grouped batching and rebasing.

### CODE REVIEW

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 1]
```diff
@@ -444,6 +464,56 @@ export function mixinReconnect<
 	};
 }
 
+/**
+ * Mixes in functionality to rebase in-flight batches in a DDS fuzz model. A batch is rebased by
```
Comment (anthony-murphy) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:468]: i have a question about what this does. Will this just rebase without interleaved changes? Another way to put it is, does this test reentrancy e.g. starting a batch during application of a remote branch?

Comment (anthony-murphy) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:468]: If it doesn't test reentrancy, does it allow for remote changes to have happened before the rebase?

What i'm trying to understand is if the state of the dds can change, due to remote changes, between submit and rebase, or will rebase be isolated to only local changes? and if there are remote changes, what flavor will they between local batch generation, or during

Comment (andre4i) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:468]: Just local changes. There is no current support in the fuzz test synchronization to interleave how batches are sent/received, currently it's all in lockstep. This is actually the first change to support batches in general (see the changes to the synchronize operation). It's doable and planned but that's not part of this change. I think I saw a task which tracks this already...

Comment (anthony-murphy) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:468]: great. glad to see it is already finding bugs. i think @Abe27342 is the right person to sign off here, but overall looks good to me.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**